### PR TITLE
PR for branch decorate_class that implements the Decorator design pattern to correct the names of people

### PR DESCRIPTION
**Pull request by jkanyi-web**
1. Created class Nameable and implemented correct_name that raises NotImplementedError

2. Turned Person class to Nameable by ensuring that the Person class inherits from Nameable

- Implemented method correct_name on person class that simply returns the name attribute

3. Prepared base_decorator Decorator and ensured it inherits from Nameable

- assign a nameable object from params to an instance variable in the constructor.
- Implemented method correct_name on person class that returns result of the correct_name method of the @nameable.

4. Prepared CapitalizeDecorator and TrimmerDecorator
- Created a CapitalizeDecorator class that inherits from the base Decorator class 
- Implement a method correct_name that capitalizes the output of @nameable.correct_name.

- Created a class TrimmerDecorator that inherits from the base Decorator class.
- Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

5. Create a main.rb file to run the decorators created and confirm the output

### CODE REVIEW

File: nameable_person.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class NameablePerson < Person
+  include Nameable
+  def correct_name
+    @name
+  end
+end
```
Comment (danmainah) [inline nameable_person.rb]: - [x] @jkanyi-web Great work adding the nameable module, however according to project requirement you are supposed to have a nameable class that has a method called correct_name that will raise a NotImplementedError, also this method doesn't inherit from the person class. Kindly consider removing the nameable module.
image

File: person.rb [hunk 2]
```diff
@@ -1,9 +1,11 @@
+require_relative 'nameable'
+
 class Person
+  include Nameable
```
Comment (danmainah) [inline person.rb]: - [x] @jkanyi-web Great work adding the correct_name method in the person class, however, this class is supposed to inherit from the `Nameable` class.
image

File: main.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+require_relative 'trimmer_decorator'
+
+person = Person.new(22, 'maximilianus')
+puts person.correct_name # Output: maximilianus
+
+capitalized_person = CapitalizeDecorator.new(person)
+puts capitalized_person.correct_name # Output: Maximilianus
+
+capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)
+puts capitalized_trimmed_person.correct_name # Output: Maximilia
```
Comment (danmainah) [inline main.rb:13]: - [ ] @jkanyi-web Great work adding the tests file, however a getting the error below when I run this file 
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @jkanyi-web , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Implement the trimmer and capitalize classes to decorate the correct name
✔️  Added tests file to test if the decoration functionality is working correctly.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by cynthiainga — APPROVED** [review]
Hi @jkanyi-web, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [x] Requested changes are made ✔️ 
- [x] There is a `Namable` class with a method 'correct_name' rasing an error ✔️ 
- [x] `Person` class inherits from `Nameable` class ✔️ 
- [x] Correct Gitflow ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**